global/scripts/mixtureModel.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+# mixtureModel <inFile> <outFile> [wbcmd] [melocmd] [wbshrtctscmd]
+# Performs Gaussian mixture modeling on precomputed ICs
+# Wrapper around melodic. Accepts nifti or cifti inputs.
+# Require Input:
+#   inFile  : file path to IC z-scores, file path including extension as string (accepts nifti or cifti files)
+#   outFile : file path to IC z-scores with Gaussian mixture modeling
+#              Can output nifti, nifti-gz, or cifti, depending on extension: .nii, .nii.gz, .dscalar.nii,
+#              Output format does not have to match input format, unless output is cifti.
+# Optional Inputs
+#   wbcmd   : workbench command, defaults to 'wb_command'
+#   melocmd : melodic command, defaults to 'melodic'
+#   wbshrtctscmd: workbench shortcuts command, defaults to 'wb_shortcuts'
+#
+# See also:
+# https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=fsl;6e85d498.1607
+# https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/MELODIC#Using_melodic_for_just_doing_mixture-modelling
+# https://www.fmrib.ox.ac.uk/datasets/techrep/tr02cb1/tr02cb1.pdf
+#
+# Created by Burke Rosen
+# 2024-07-08
+# Dependencies: workbench, FSL
+# Written with workbench 2.0 and FSL 6.0.7.1
+# Refactored from matlab to bash 2025-05-19 by Copilot GPT-4.1
+#
+# ToDo:
+# Feed arbitrary melodic arguments.
+
+mixtureModel() {
+  local inFile="$1"
+  local outFile="$2"
+  local wbcmd="${3:-wb_command}"
+  local melocmd="${4:-melodic}"
+  local wbshrtctscmd="${5:-wb_shortcuts}"
+
+  # handle inputs
+  if [[ -z "$inFile" || -z "$outFile" ]]; then
+    echo "inFile and outFile arguments are required!" >&2
+    return 1
+  fi
+
+  # check dependencies
+  command -v "$wbcmd" >/dev/null 2>&1 || { echo "workbench_command binary $wbcmd not on path" >&2; return 1; }
+  command -v "$wbshrtctscmd" >/dev/null 2>&1 || { echo "workbench_command binary $wbshrtctscmd not on path" >&2; return 1; }
+  command -v "$melocmd" >/dev/null 2>&1 || { echo "melodic command binary $melocmd not on path" >&2; return 1; }
+
+  local inFile0="$inFile"
+  local outFile0="$outFile"
+  local tDir
+  tDir=$(mktemp -d)
+  local FSLOUTPUTTYPE0="$FSLOUTPUTTYPE"
+  local stat=0
+  local ext
+
+  # convert input if needed
+  if [[ "$inFile0" == *dscalar.nii ]]; then
+    # convert input cifti to nifti
+    inFile="$(mktemp --suffix=.nii)"
+    $wbcmd -cifti-convert -to-nifti "$inFile0" "$inFile" -smaller-dims
+    # local dims
+    dims=$($wbcmd -file-information "$inFile" | grep Dimensions | awk -F': ' '{print $2}')
+    # shellcheck disable=SC2206
+    dims_arr=($dims)
```

Comment:
The `-file-information` output uses commas and spaces, while bash would need the comma removed.

It would be a little safer to use `grep '^Dimensions' to specifically require it to match the start of the line (if `-no-map-info` isn't used, and some goofball names one of their maps "Dimensions"...).

I generally recommend `-no-map-info` on `-file-information`, particularly for scripting.  On cifti files it avoids loading the data, but volume files don't support not loading the data, so it is mostly cutting down the amount of stuff grep might see.

`-file-information` probably won't give you a "1" for the number of maps if there is only one map.  Maybe that isn't the error condition you are looking for, though.

```suggestion
    "$wbcmd" -cifti-convert -to-nifti "$inFile0" "$inFile" -smaller-dims
    # local dims
    dims=$("$wbcmd" -file-information "$inFile" | grep Dimensions | awk -F': ' '{print $2}' | tr ',' ' ')
    # shellcheck disable=SC2206
    dims_arr=($dims)
```

Reply:
A couple of fslvals is  probably the best solution. I guess I digitized what I as a human would literally do if I were checking manually ( -file-information  and look at the right spot in the output).